judge/views/api/api_v2.py

```diff
@@ -527,6 +552,13 @@ def get_object_data(self, profile):
 
         last_rating = profile.ratings.order_by('-contest__end_time').first()
 
+        new_ratings_subquery = Rating.objects.filter(participation=OuterRef('pk'))
+        old_ratings_subquery = (
+            Rating.objects
+            .filter(user=OuterRef('user__pk'), contest__end_time__lt=OuterRef('contest__end_time'))
+            .order_by('-contest__end_time')
+        )
+
         contest_history = []
         participations = (
             ContestParticipation.objects
```

Comment:
Do we want to `select_related('contest')`?

Reply:
If so, should https://github.com/JoshuaTianYangLiu/online-judge/blob/api-consistency/judge/views/api/api_v2.py#L256 be changed as well?

Reply:
Yes, I think it should be changed as well.